app/uk/gov/hmrc/economiccrimelevystubs/services/EclRegistrationReferenceService.scala

```diff
@@ -34,4 +35,14 @@ class EclRegistrationReferenceService @Inject() (sequenceRepository: SequenceRep
       s"$prefix${"%010d".format(id)}"
     }
 
+  def getPreviousReferences: Future[List[String]] =
+    sequenceRepository.getCurrentReference(eclRegistrationReferenceKey).map {
+      case Some(ref) =>
+        val referenceRange = List.range(1, ref + 1)
+        val references     = new ListBuffer[String]()
+        for (elem <- referenceRange)
+          references += s"$prefix${"%010d".format(elem)}"
+        references.toList
+      case _         => List.empty
+    }
```

Comment:
We tend not to favour side effecting code like this in Scala, e.g. using a for loop to build up a mutable list. 

For something like this you can use `map` which will apply a function to every element in the list and produce a new resulting List, i.e. `List[A] => List[B]`

```suggestion
      def getPreviousReferences: Future[List[String]] =
    sequenceRepository
      .getCurrentReference(eclRegistrationReferenceKey)
      .map(_.map(ref => List.range(1, ref + 1).map(i => s"$prefix${"%010d".format(i)}")).toList.flatten)
```

or if you prefer to keep the pattern match on the option:

```suggestion
      def getPreviousReferences: Future[List[String]] =
    sequenceRepository.getCurrentReference(eclRegistrationReferenceKey).map {
      case Some(ref) => List.range(1, ref + 1).map(i => s"$prefix${"%010d".format(i)}")
      case _         => List.empty
    }
```

Reply:
Done. Thank you for suggesting this. It helps a lot to accommodate to new way of thinking and writing😇